src/components/MapsSection/MapsSection.tsx

```diff
@@ -102,6 +104,57 @@ const MapsSection = ({
     });
   }, [currentVisu]);
 
+  useEffect(() => {
+    const button = buttonRef.current;
+    const container = tagsContainerRef.current;
+
+    if (container && button) {
+      const currentIndex = tiffInfo.findIndex(
+        (map) => map.fields.id === currentVisu.id,
+      );
+
+      const containerTop = container.scrollTop;
+      const containerBottom = containerTop + container.clientHeight;
+
+      const buttonOffsetHeight = button.offsetHeight;
+      const buttonHeight = button.offsetTop;
+      +buttonOffsetHeight;
+
+      let tamBotao = buttonHeight / 100;
+      const isMobile = window.innerWidth <= 810;
+      const screen = window.innerHeight;
+      const maxVisibleBoxes = isMobile ? 1 : 5;
+      const needsScroll =
+        (buttonHeight / 9) * currentIndex > containerBottom ||
+        currentIndex === 0;
+
+      if (needsScroll) {
+        const scrollDesktop = Math.abs(
+          tamBotao + (currentIndex - maxVisibleBoxes) * buttonOffsetHeight,
```

Comment:
How did you came up with this solution? Just curious

Reply:
Since I couldn’t find a function that would return the exact position of a button relative to the container and considering overflow, I started observing some patterns. The scroll will always happen based on **the size of the button I’m scrolling to** (tamBotao) (tamBotao is different from buttonOffsetHeight because tamBotao considers the size of the open button, while buttonOffsetHeight is the size of the closed button) **+ the number of closed buttons behind it** ((currentIndex - maxVisibleBoxes) * buttonOffsetHeight). The need for maxVisibleBoxes is because it ensures scrolling only considers buttons that aren’t overflowing above the container; without this variable, it would multiply by the total number of buttons behind it, including those overflowing, which would make the calculation incorrect